apollo-router-core/src/services/router_service.rs

```diff
@@ -308,14 +309,18 @@ impl PluggableRouterServiceBuilder {
         let (router_service, router_worker) = Buffer::pair(
             ServiceBuilder::new().service(
                 self.plugins.iter_mut().fold(
-                    RouterService::builder()
-                        .query_planner_service(query_planner_service)
-                        .query_execution_service(execution_service)
-                        .schema(self.schema)
-                        .query_cache(query_cache)
-                        .naive_introspection(naive_introspection)
-                        .build()
-                        .boxed(),
+                    // TODO: maybe make it optional although enabled by default?
+                    APQService::new(
+                        RouterService::builder()
+                            .query_planner_service(query_planner_service)
+                            .query_execution_service(execution_service)
+                            .schema(self.schema)
+                            .query_cache(query_cache)
+                            .naive_introspection(naive_introspection)
+                            .build(),
+                        512, // TODO: this is totally arbitrary, what does the team think?
```

Comment:
I like multiples of 2! :) More seriously: new configuration item?

Reply:
we'll figure it out in #440 then!